src/CqlBuilderPanel/definitionsSection/DefinitionsSection.tsx

```diff
@@ -1,29 +1,162 @@
+  const [currentLimit, setCurrentLimit] = useState<number>(5);
+  const [currentPage, setCurrentPage] = useState<number>(1);
+
+  const canGoPrev = currentPage > 1;
+  const canGoNext = (() => {
+    return currentPage < totalPages;
+  })();
+  const handlePageChange = (e, v) => {
+    setCurrentPage(v);
+  };
+  const handleLimitChange = (e) => {
+    setCurrentLimit(e.target.value);
+    setCurrentPage(1);
+  };
+  useEffect(() => {
+    managePagination();
+  }, [cqlBuilderLookupsTypes, currentPage, currentLimit]);
+
+  // table data
+  const data = visibleDefinitions?.map((definition) => {
+    return {
+      name: definition.name,
+      comment: definition.comment,
+    } as unknown as Definition;
+  });
+  const columns = useMemo<ColumnDef<Definition>[]>(
+    () => [
+      {
+        header: "Name",
+        accessorKey: "name",
+      },
+      {
+        header: "Comment",
+        accessorKey: "comment",
+      },
+      {
+        header: "",
+        accessorKey: "apply",
+        cell: (row: any) => {
+          return (
+            <>
+              <DeleteOutlineOutlinedIcon htmlColor="red" />
+              <EditNoteRoundedIcon color="primary" />
+            </>
+          );
+        },
+      },
+    ],
+    [cqlBuilderLookupsTypes]
+  );
+
+  const table = useReactTable({
+    data,
+    columns,
+    getCoreRowModel: getCoreRowModel(),
+  });
+
+  const managePagination = useCallback(() => {
+    let definitions;
+    if (cqlBuilderLookupsTypes?.definitions) {
+      definitions = cqlBuilderLookupsTypes?.definitions.map((definition) => {
+        return {
+          name: definition,
+          comment: "",
+        } as unknown as Definition;
+      });
+    } else {
+      definitions = {} as unknown as Definition;
+    }
+
+    if (definitions?.length > 0) {
+      setTotalItems(definitions.length);
+      if (definitions.length < currentLimit) {
+        setOffset(0);
+        setVisibleDefinitions(definitions && [...definitions]);
+        setVisibleItems(definitions?.length);
+        setTotalPages(1);
+      } else {
+        const start = (currentPage - 1) * currentLimit;
+        const end = start + currentLimit;
+        const newVisibleCodes = [...definitions].slice(start, end);
+        setOffset(start);
+        setVisibleDefinitions(newVisibleCodes);
+        setVisibleItems(newVisibleCodes?.length);
+        setTotalPages(Math.ceil(definitions?.length / currentLimit));
+      }
+    }
+  }, [
+    currentLimit,
+    currentPage,
+    cqlBuilderLookupsTypes,
+    setOffset,
+    setVisibleDefinitions,
+    setVisibleItems,
+    setTotalItems,
+    setTotalPages,
+  ]);
+
   return (
     <>
       <DefinitionSectionNavTabs
         activeTab={activeTab}
         setActiveTab={setActiveTab}
+        definitionCount={(() => {
+          if (cqlBuilderLookupsTypes?.definitions) {
+            return cqlBuilderLookupsTypes.definitions.length;
+          } else {
+            return 0;
+          }
```

Comment:
I think we can directly write conditional render here instead of if else. conditional render would simplify the code syntax

definitionCount= {cqlBuilderLookupsTypes?.definitions ? cqlBuilderLookupsTypes?.definitions.length: 0}